p2p/host/basic/basic_host.go

```diff
@@ -366,10 +359,36 @@ func (h *BasicHost) background(p goprocess.Process) {
 	}
 	h.mx.Unlock()
 
+	//  emit an EvtLocalAddressesUpdatedEvent & a Push Identify if our listen addresses have changed.
+	go func() {
+		for {
+			select {
+			case <-h.addrChangeChan:
```

Comment:
can you not incorporate this into the below for() loop on the background goroutine, rather than adding the additional event loop for translating this event?

Reply:
Good idea, have made the change.